sklearn/utils/validation.py

```diff
@@ -1999,6 +1999,45 @@ def _is_pandas_df(X):
     return False
 
 
+def _is_polars_df(X):
+    """Return True if the X is a polars dataframe."""
+    if hasattr(X, "columns") and hasattr(X, "schema"):
+        # Likely a polars DataFrame, we explicitly check the type to confirm.
+        try:
+            pl = sys.modules["polars"]
+        except KeyError:
+            return False
+        return isinstance(X, pl.DataFrame)
+    return False
+
+
+def _dataframe_module_as_str(df, estimator=None):
+    """Return the dataframe module of df as a string"""
+    estimator_name = _check_estimator_name(estimator)
+    context = " by %s" % estimator_name if estimator is not None else ""
+    if _is_pandas_df(df):
+        return "pandas"
+    elif _is_polars_df(df):
+        return "polars"
+    else:
+        raise ValueError(f"Only Pandas and Polars dataframes are supported {context}")
+
+
+def _interchange_to_dataframe(df_interchange, to_dataframe_library):
+    """Converts to DataFrame using interchange protocol.
+
+    Only pandas and polars are supported.
+    """
+    if to_dataframe_library == "pandas":
+        import pandas as pd
+
+        return pd.api.interchange.from_dataframe(df_interchange)
```

Comment:
Functionally, this code path is not being used in `ColumnTransformer`, so I think it is okay. 

`ColumnTransformer` is still special causing pandas dataframes because of how the `__dataframe__` interchange protocol can sometimes make a copy.